src/runtime_shared.js

```diff
@@ -81,8 +81,64 @@ function updateMemoryViews() {
   {{{ maybeExportHeap('HEAP64')  }}}HEAP64 = new BigInt64Array(b);
   {{{ maybeExportHeap('HEAPU64') }}}HEAPU64 = new BigUint64Array(b);
 #endif
+
+#if SUPPORT_BIG_ENDIAN
+   HEAPU16.unsigned = (x => x >>> 0);
+   HEAPU32.unsigned = (x => x >>> 0);
+#if WASM_BIGINT
+   HEAPU64.unsigned = (x => x >= 0 ? x : BigInt(2**64) + x);
+#endif
```

Comment:
Ideally we could avoid adding this here and keep all the code chagnes to liblittle_endian_heap.js.

Ideally we could also avoid adding extra methods to these heaps, but maybe thats not easy.

Reply:
I didn't really find a better way to keep the extra methods out while not introducing per-type functions (and associated transformations).

Reply:
Can you just use the uppercase version here?    i.e. why do we need the new lower case versions?

Reply:
I wonder if wen also move these lines to liblittle_endian_heap.js?

Perhaps they could be part of the `__postset` too?

Reply:
re the new variables - if the uppercased variables are used they get replaced by e.g. the growable heap wrappers.

Reply:
One option would be to rename `updateMemoryViews` to `_updateMemoryViews`. Assign `_updateMemoryViews` to `updateMemoryViews` and use to `__postset` to override the new `updateMemoryViews` variable.

If you prefer this option, I can look into that on Monday.

Reply:
Regarding the `HEAPXX` variable substitution, we do have a list of function for which this is not performed.  I don't know if that will be useful in this case but the list is here: https://github.com/emscripten-core/emscripten/blob/cb8d502aa7213167bc773aab9c10d9666dd52c92/tools/acorn-optimizer.mjs#L1484-L1491